In [1]:
import math
import scipy.stats as stats

# Step 1: Define the parameters from your data and cost model
sample_mean = 3050  # Observed sample mean weekly cost (Rs.)
sample_size = 25    # Number of restaurants in the sample
X_mean = 600        # Mean number of units produced per week
model_sigma_units = 25  # Std. dev. of units produced
sigma_cost_per_unit = 5 # Model cost per unit

# Theoretical weekly operating cost (mean)
theoretical_mean = 1000 + (5 * X_mean)

# Theoretical weekly cost standard deviation
sigma = sigma_cost_per_unit * model_sigma_units

# Step 2: Calculate the test statistic
z_stat = (sample_mean - theoretical_mean) / (sigma / math.sqrt(sample_size))

# Step 3: Find the critical z-value for 5% significance level (one-tailed test, right side)
alpha = 0.05
z_critical = stats.norm.ppf(1 - alpha)

# Step 4: Decision
reject_null = z_stat > z_critical

# Step 5: Print results and conclusion
print("Sample Mean (Rs.):", sample_mean)
print("Theoretical Mean (Rs.):", theoretical_mean)
print("Sample Size:", sample_size)
print("Standard Deviation (Rs.):", sigma)
print("Test Statistic (z):", z_stat)
print("Critical Value (z) at α=0.05:", z_critical)
print("Reject Null Hypothesis:", reject_null)
if reject_null:
    print("Conclusion: There is strong evidence that weekly operating costs are higher than the model suggests.")
else:
    print("Conclusion: There is NOT strong evidence that weekly operating costs are higher than the model suggests.")


Sample Mean (Rs.): 3050
Theoretical Mean (Rs.): 4000
Sample Size: 25
Standard Deviation (Rs.): 125
Test Statistic (z): -38.0
Critical Value (z) at α=0.05: 1.6448536269514722
Reject Null Hypothesis: False
Conclusion: There is NOT strong evidence that weekly operating costs are higher than the model suggests.


In [2]:
import numpy as np
import scipy.stats as stats


In [3]:
# Durability measurements (millions of characters)
durability = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])
n = len(durability)


In [4]:
sample_mean = np.mean(durability)
sample_std = np.std(durability, ddof=1)
print("Sample Mean:", sample_mean)
print("Sample Standard Deviation:", sample_std)
print("Sample Size:", n)


Sample Mean: 1.2386666666666666
Sample Standard Deviation: 0.19316412956959936
Sample Size: 15


In [5]:
confidence = 0.99
df = n - 1
t_critical = stats.t.ppf(1 - (1-confidence)/2, df=df)
margin_of_error_t = t_critical * (sample_std / np.sqrt(n))
ci_t_lower = sample_mean - margin_of_error_t
ci_t_upper = sample_mean + margin_of_error_t
print(f"99% Confidence Interval (t-distribution): ({ci_t_lower:.2f}, {ci_t_upper:.2f})")


99% Confidence Interval (t-distribution): (1.09, 1.39)


In [6]:
pop_std = 0.2  # million characters, known population standard deviation
z_critical = stats.norm.ppf(1 - (1-confidence)/2)
margin_of_error_z = z_critical * (pop_std / np.sqrt(n))
ci_z_lower = sample_mean - margin_of_error_z
ci_z_upper = sample_mean + margin_of_error_z
print(f"99% Confidence Interval (z-distribution): ({ci_z_lower:.2f}, {ci_z_upper:.2f})")


99% Confidence Interval (z-distribution): (1.11, 1.37)


#### The analysis estimates the average durability of print-heads measured by the number of characters printed before failure, based on a sample of 15 print-heads tested destructively.

Two 99% confidence intervals were constructed to estimate the true mean durability:

Using the sample standard deviation and the t-distribution (appropriate for small samples and unknown population variance), the interval is approximately (1.09, 1.39) million characters. This interval accounts for extra uncertainty from estimating variability from the sample.

Assuming the population standard deviation is known (0.2 million characters), the z-distribution was used, resulting in a slightly narrower interval of about (1.11, 1.37) million characters.

Both intervals suggest with 99% confidence that the true mean durability lies within roughly 1.1 to 1.4 million characters printed before failure. The t-based interval is wider due to uncertainty in estimating the population standard deviation from a small sample.

This analysis provides a statistically sound range for the manufacturer’s mean print-head durability, useful for quality control and production decisions.


## Chi-Square Test of Independence

### Objective
To determine whether there is a significant association between two categorical variables using the Chi-Square test of independence.

### Hypotheses
- **Null Hypothesis (H₀):** There is no association between the categorical variables.
- **Alternative Hypothesis (H₁):** There is a significant association between the categorical variables.

### Significance Level
α = 0.05


In [ ]:

# Chi-Square Test Implementation

import pandas as pd
from scipy.stats import chi2_contingency

# Example contingency table (replace with assignment-specific data if required)
data = {
    'Category_A': [20, 30, 25],
    'Category_B': [22, 28, 35]
}

contingency_table = pd.DataFrame(data, index=['Group1', 'Group2', 'Group3'])
contingency_table


In [ ]:

# Perform Chi-Square Test
chi2, p, dof, expected = chi2_contingency(contingency_table)

print(f"Chi-Square Statistic: {chi2}")
print(f"P-value: {p}")
print(f"Degrees of Freedom: {dof}")

# Conclusion
if p < 0.05:
    print("Reject the Null Hypothesis: There is a significant association between variables.")
else:
    print("Fail to Reject the Null Hypothesis: No significant association between variables.")



### Conclusion
Based on the Chi-Square test results and the p-value comparison with the significance level,
we conclude whether an association exists between the categorical variables.
